### PR TITLE
Backend initialization and endpoints for users and ice-creams

### PR DESCRIPTION
**Pull request by meru-a**
Initialized backend and created first endpoints for issues #35 #37

### CODE REVIEW

File: server/controllers/users.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    const hashedpassword = await argon2.hash(password)
+
+    const user = new User({
+      username: username,
+      password: hashedpassword,
+    })
+
+    console.log(user)
```
Comment (jgesel) [inline server/controllers/users.ts:28]: delete this console.log

File: server/index.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+const port = PORT;
+
+app.use('/api/icecreams', icecreamRouter);
+app.use('/api/users', userRouter);
+
+const start = async () => {
+  try {
+    await dbConnect()
+    console.log("connected to DB")
+    app.listen(port, () => console.log("Server started on port 3000"));
```
Comment (jgesel) [inline server/index.ts:17]: switch to formatted string port `on port ${port}`

File: server/lib/dbConnect.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import mongoose from 'mongoose'
+import { MONGODB_URI } from '../utils/config'
+
+declare global {
+  var mongoose: any // This must be a `var` and not a `let / const`
```
Comment (jgesel) [inline server/lib/dbConnect.ts:5]: is this supposed to be type 'any' or is there a specific type we want?

**Review by jgesel — APPROVED** [review]
Just a few small comments, otherwise looks good!

File: server/index.ts (hunk 2 above)
Comment (meru-a) [reply server/index.ts]: Good catch, changed it!

File: server/lib/dbConnect.ts (hunk 3 above)
Comment (meru-a) [reply server/lib/dbConnect.ts:5]: No idea, tbh. I took the Mongoose connection approach from how [Next does it here](https://github.com/vercel/next.js/blob/canary/examples/with-mongodb-mongoose/lib/dbConnect.ts)

**Pull request merged**